### PR TITLE
Run scala:doc-jar on PRs along with javadoc task

### PR DESCRIPTION
**Pull request by nastra**
We had PRs fail after being merged to main because the `doc-jar` task
failed (since it is only part of the `release` profile).
However, given that we already run `javadoc` on PRs, we should do the
same for scala.

### CODE REVIEW

**Review by XN137 — APPROVED** [review]
+1 since it makes sense to prevent breaking the main build
... hoping there are no negative sideeffects i.e. much longer CI run times :sweat_smile:

**Review by dimas-b — APPROVED** [review]

**Comment by snazy** [comment]
@nastra wanna get this done?

**Comment by nastra** [comment]
> @nastra wanna get this done?

no resources atm to take care of this :(

File: clients/deltalake/pom.xml [hunk 1]
```diff
@@ -308,6 +308,9 @@
                 <goals>
                   <goal>doc-jar</goal>
                 </goals>
+                <configuration>
+                  <scalaVersion>${scala2.12.version}</scalaVersion>
```
Comment (XN137) [inline clients/deltalake/pom.xml:312]: does it not warrant a comment why only for this usage of the plugin we have to use a specific version and in the others we dont?

File: pom.xml [hunk 2]
```diff
@@ -818,7 +818,6 @@
           <version>4.6.1</version>
           <configuration>
             <sendJavaToScalac>false</sendJavaToScalac>
-            <scalaVersion>${scala2.12.version}</scalaVersion>
```
Comment (XN137) [inline pom.xml:821]: what version will be used now that this is no longer specified?
was this always wrong or how is this related now to generating scala doc?

**Review by XN137 — COMMENTED** [review]
theoretically +1 since CI is green i guess but i dont really understand it tbh (see questions)

File: pom.xml (hunk 2 above)
Comment (nastra) [reply pom.xml:821]: setting this version globally resulted in a failure when the gatling module was executed. All of the other modules (except deltalake were already specifiying scala 2.12). Only the gatling module requires scala 2.13 but as I said, setting the version globally here resulted in issues

File: clients/deltalake/pom.xml (hunk 1 above)
Comment (nastra) [reply clients/deltalake/pom.xml:312]: all of the other scala modules were already explicitly using scala 2.12, this is just aligning the deltalake module
